src/componentize.js

```diff
@@ -181,42 +183,80 @@ export async function componentize(jsSource, witWorld, opts) {
     console.log('--- Wizer Env ---');
     console.log(env);
   }
-
-  try {
-    let wizerProcess = spawnSync(
-      wizer,
-      [
-        '--allow-wasi',
-        '--init-func',
-        'componentize.wizer',
-        `--dir=${maybeWindowsPath(sourceDir)}`,
-        `--wasm-bulk-memory=true`,
-        '--inherit-env=true',
-        `-o=${output}`,
-        input,
-      ],
-      {
-        stdio: [null, stdout, stderr],
-        env,
-        input: maybeWindowsPath(
-          join(sourceDir, sourceName.slice(0, -3) + '.bindings.js')
-        ),
-        shell: true,
-        encoding: 'utf-8',
+  if (opts.enableAot) {
+    const wevalBin = await getWeval();
+
+    try {
+      let wevalProcess = spawnSync(
+        wevalBin,
+        [
+          'weval',
+          `--cache-ro ${aotCache}`,
+          `--dir ${maybeWindowsPath(sourceDir)}`,
+          '-w',
+          `-i ${input}`,
+          `-o ${output}`
+        ],
+        {
+          stdio: [null, stdout, stderr],
+          env,
+          input: maybeWindowsPath(
+            join(sourceDir, sourceName.slice(0, -3) + '.bindings.js')
+          ),
+          shell: true,
+          encoding: 'utf-8',
+        }
+      );
+      if (wevalProcess.status !== 0)
+        throw new Error('Wevaling failed to complete');
+    } catch (error) {
+      let err =
+        `Failed to initialize the compiled Wasm binary with Weval:\n` +
+        error.message;
+      if (DEBUG_BINDINGS) {
+        err += `\nBinary and sources available for debugging at ${tmpDir}\n`;
+      } else {
+        rmSync(tmpDir, { recursive: true });
       }
-    );
-    if (wizerProcess.status !== 0)
-      throw new Error('Wizering failed to complete');
-  } catch (error) {
-    let err =
-      `Failed to initialize the compiled Wasm binary with Wizer:\n` +
-      error.message;
-    if (DEBUG_BINDINGS) {
-      err += `\nBinary and sources available for debugging at ${tmpDir}\n`;
-    } else {
-      rmSync(tmpDir, { recursive: true });
+      throw new Error(err);
+    }
+  } else {
+    try {
+      let wizerProcess = spawnSync(
+        wizer,
+        [
+          '--allow-wasi',
+          '--init-func',
+          'componentize.wizer',
```

Comment:
`--init-func` must be provided.

Reply:
```
error: Found argument '--init-func' which wasn't expected, or isn't valid in this context
```

These are not arguments that are passed in the reference I was using from the [fastly js-compute-runtime diff](https://github.com/fastly/js-compute-runtime/pull/883/files)

Reply:
`--init-func` is a Wizer option ComponentizeJS needs to use, and Weval accepts common Wizer options.

Reply:
Ahh Weval doesn't support this so we need to either work around this custom init function by using the common Wizer name here (updating `engine.cpp` to use `wizer.initialize`), or we need to get this option supported in Weval.

Reply:
Would it be a possibility to run `wizer` first then followed by running `weval` without `-w` flag

Reply:
It should be relatively straightforward to add this option to Weval and pipe it through to Wizer here - https://github.com/cfallin/weval/blob/main/src/main.rs#L94.

Would you be able to look into this for Weval further?

Reply:
Yep - I just started looking at it. Will open a PR soon.

Reply:
Chris is already on it - https://github.com/cfallin/weval/commit/81226fd1a23ab446be31330d5fb55ec387ffe101, so next Weval release should be good to support this.

Reply:
Oh, that was fast! Thanks!

Reply:
This should be in Weval 0.2.13 now.

Reply:
Just pushed the changes and the test seems to pass locally so far!